decidim-core/app/controllers/decidim/locales_controller.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+require_dependency "decidim/application_controller"
+
+module Decidim
+  # A controller to allow users switching their locale.
+  class LocalesController < ApplicationController
+    def create
+      if current_user && params["locale"] && available_locales.include?(params["locale"])
+        current_user.update_attribute(:locale, params["locale"])
+      end
+
+      uri = URI(params["back"] || "/")
+      params = URI.decode_www_form(uri.query || "") << ["locale", current_locale]
```

Comment:
Exactly the same issue, the interface language would change as well. I prefer having a `locale` param over the prepended route, at least it's the same resource.

I know there's controversy on how to deal with locales. What I'm suggesting might not be optimal for SEO though. Let's keep the current approach and think about this later.

**Note**: We should include GET links somewhere to switch languages if we want to really make it SEO-friendly.